test/integration/integration_admin_test.cc

```diff
@@ -161,10 +161,37 @@ TEST_P(IntegrationAdminTest, Admin) {
   response = IntegrationUtil::makeSingleRequest(lookupPort("admin"), "GET", "/stats?format=json",
                                                 "", downstreamProtocol(), version_);
   EXPECT_TRUE(response->complete());
+  EXPECT_STREQ("application/json", ContentType(response));
   EXPECT_STREQ("200", response->headers().Status()->value().c_str());
+
   Json::ObjectSharedPtr statsjson = Json::Factory::loadFromString(response->body());
   EXPECT_TRUE(statsjson->hasObject("stats"));
-  EXPECT_STREQ("application/json", ContentType(response));
+
+  bool has_histograms = false;
+  for (Json::ObjectSharedPtr obj_ptr : statsjson->getObjectArray("stats")) {
+    if (obj_ptr->hasObject("histograms")) {
+      has_histograms = true;
+      const Json::ObjectSharedPtr& histograms_ptr = obj_ptr->getObject("histograms");
+      // Validate that both supported_quantiles and computed_quantiles are present in JSON.
+      EXPECT_TRUE(histograms_ptr->hasObject("supported_quantiles"));
+      EXPECT_TRUE(histograms_ptr->hasObject("computed_quantiles"));
+
+      const std::vector<Json::ObjectSharedPtr>& computed_quantiles =
+          histograms_ptr->getObjectArray("computed_quantiles");
+      EXPECT_GT(computed_quantiles.size(), 0);
+
+      // Validate that each computed_quantile has name and value objects.
+      EXPECT_TRUE(computed_quantiles[0]->hasObject("name"));
+      EXPECT_TRUE(computed_quantiles[0]->hasObject("values"));
+
+      // Validate that supported and computed quantiles are of the same size.
+      EXPECT_EQ(histograms_ptr->getObjectArray("supported_quantiles").size(),
+                computed_quantiles[0]->getObjectArray("values").size());
+    }
+  }
+
+  // Validate that the stats JSON has histograms.
+  EXPECT_TRUE(has_histograms);
```

Comment:
+1

Reply:
I made it to bool because, we will only have one "histograms" top level object in the JSON which  has supported_quantlies (one) and an array of computed_quantiles for each histogram. I may be missing some thing here and not able to follow your suggestion about counting.